File: Quiz-server/src/main/java/com/qucat/quiz/repositories/entities/Question.java
package com.qucat.quiz.repositories.entities;

import lombok.Builder;
import lombok.Data;

import java.util.List;

@Data
@Builder
public class Question {
    private int id;
    private int quizId;
    private QuestionType type;
    private String content;
    private int score;
    private int imageId;
    private Image image;
    private List<QuestionOption> options;
}

File: Quiz-web/src/app/modules/core/models/question.ts
import {QuestionType} from './questionType';
import {Imaged} from "./imaged";
import {QuestionOptions} from "./questionOptions";
import {Quiz} from "./quiz";
import {Image} from "./image";

export class Question implements Imaged {
  id: number;
  type: QuestionType = QuestionType.SELECT_OPTION;
  content: string;
  score: number;
  imageId: number = -1;
  options: QuestionOptions[] = [];
  image: Image = new Image();

}
